src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -0,0 +1,203 @@
+package partitions
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-units"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/types"
+)
+
+func newListPartitionMovementsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		completion int
+		all        bool
+		human      bool
+		partitions []string
+		response   []adminapi.ReconfigurationsResponse
+	)
+	cmd := &cobra.Command{
+		Use:     "move-status",
+		Aliases: []string{"alter-status"},
+		Short:   "Show ongoing partition movements",
+		Long:    helpListMovement,
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			// If partition(s) is specified but no topic(s) is specified, exit.
+			if len(topics) <= 0 && len(partitions) > 0 {
+				out.Die("specify at least one topic when --partition is used, exiting.")
+			}
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			response, err = cl.Reconfigurations(cmd.Context())
+			out.MaybeDie(err, "unable to list partition movements: %v\n", err)
+
+			if len(response) == 0 {
+				out.Exit("There are no ongoing partition movements.")
+			}
+
+			j := 0
+			for _, t := range topics {
+				nt := strings.Split(t, "/")
+				if len(nt) > 2 {
+					fmt.Printf("invalid format for topic %s, skipping.\n", t)
+					continue
+				}
+				for _, r := range response {
+					isKafkaNs := len(nt) == 1 && r.Ns == "kafka" && r.Topic == t
+					isInternalNs := len(nt) == 2 && r.Ns == nt[0] && r.Topic == nt[1]
+
+					if isKafkaNs || isInternalNs {
+						if len(partitions) == 0 || contains(partitions, strconv.Itoa(r.PartitionID)) {
+							response[j] = r
+							j++
+						}
+					}
+				}
+			}
+			if j > 0 {
+				response = response[:j]
+			}
```

Comment:
To avoid modifying the original response (L63) what do you think creating a new variable for the filteredResponses and dropping the `j`?

```
var filteredResponses []<type>
for _, t := range topics{
...
    if len(partitions) == 0 || contains(partitions, strconv.Itoa(r.PartitionID)) {
		filteredResponse = append(filteredResponse, r)
    }
```

Will that work?

Reply:
Sure, that'll be easier to understand what it's doing. Thank you.